### PR TITLE
[Swift5] small improvements to Objc compatibility

### PR DESCRIPTION
**Pull request by 4brunu**
The Objc compatibility is broken, but this will help to improve things a bit.

This PR adds a new objc reserved word, and instead of vars, it processes allVars, because with vars, some variables were not getting the vendorExtensions `x-swift-optional-scalar` tag.

### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [X] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [X] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@jgavris (2017/07) @ehyche (2017/08) @Edubits (2017/09) @jaz-ah (2017/09) @4brunu (2019/11)

### CODE REVIEW

**Comment by 4brunu** [comment]
@wing328 I'm fighting the Swift4 CI.
Does this makes sense at this point that we have the Swift 5 generator?
Should we disable Swift4 from CI?

**Comment by wing328** [comment]
> @wing328 I'm fighting the Swift4 CI.

It could be due to bad data in the public petstore server. I will see if I can apply the same workaround that I used for swift 5 client.

**Comment by 4brunu** [comment]
Hum, sorry to bother you, I think I found the problem

**Comment by 4brunu** [comment]
Finally CI passed.
For me this is ready to merge.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java [hunk 1]
```diff
@@ -2252,6 +2252,11 @@ public CodegenModel fromModel(String name, Schema schema) {
                 postProcessModelProperty(m, prop);
             }
         }
+        if (m.allVars != null) {
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java]: I may add a comment later to explain the difference between vars and allVars.

**Pull request merged**